storage/storage_reference.go

```diff
@@ -30,6 +33,8 @@ func newReference(transport storageTransport, reference, id string, name referen
 		reference: reference,
 		id:        id,
 		name:      name,
+		tag:       tag,
+		digest:    digest,
```

Comment:
Why aren’t the tag and digest values stored inside the `reference.Named`? That would eliminate the failure paths in `DockerReference` which now have to be silently ignored.